### ISSUE TITLE
Getting error while copying data from Oracle to Postgres

### ISSUE THREAD

**Issue by jitenpansara** [darold/ora2pg#1238]
![postgres_copy_error](https://user-images.githubusercontent.com/53655001/129933499-4e558e67-e020-415d-a38c-5c1d40b07fe4.JPG)
Hello Team,

We have successfully created DDL statements of all objects but while copying, we are getting error as attached in the error. Requesting you to help.

C:\ora2pg\migration\test_project>ora2pg -c C:\ora2pg\migration\test_project\config\ora2pg.conf -t COPY --pg_dsn "dbi:Pg:dbname=postgres;host=localhost;port=5432" --pg_user postgres
[========================>] 804/804 tables (100.0%) end of scanning.
DBD::Pg::db do failed: ERROR:  relation "facility" does not exist at C:/Strawberry/perl/site/lib/Ora2Pg.pm line 8182.
FATAL: ERROR:  relation "facility" does not exist
Aborting export...

Regards,
Jiten

**Comment by darold** [comment]
Hi Jiten,

I don't think you have imported all DDL into the PostgreSQL database or at least not in the same schema where the data are imported. You must check in what PG schema the tables have been created and set PG_SCHEMA accordingly. If they are in the public schema, do not set PG_SCHEMA and disable EXPORT_SCHEMA.

Best regards,

**Comment by jitenpansara** [comment]
Really appreciate your quick response.

Yes, i do agree. I have fixed it by creating the missing object. But now i am getting below error.

[========================>] 804/804 tables (100.0%) end of scanning.
DBD::Pg::db do failed: ERROR:  cannot truncate a table referenced in a foreign key constraint
DETAIL:  Table "asset_custom_flow" references "accessory".
HINT:  Truncate table "asset_custom_flow" at the same time, or use TRUNCATE ... CASCADE. at C:/Strawberry/perl/site/lib/Ora2Pg.pm line 3905.
FATAL: ERROR:  cannot truncate a table referenced in a foreign key constraint
DETAIL:  Table "asset_custom_flow" references "accessory".
HINT:  Truncate table "asset_custom_flow" at the same time, or use TRUNCATE ... CASCADE.

I have modified data files from **C:\ora2pg\migration\test_project\data** location, used Cascade and even tried to disable the constraints but still getting the same error. Any suggestions?

Regards,
Jiten

**Comment by darold** [comment]
Yes this is because you have created the constraints before importing the data, enable DROP_FKEY in ora2pg.conf, it will drop the FK before importing the data and recreates them ar end.

**Comment by jitenpansara** [comment]
Sure...Will do that. It's too late now, will let you know in case of any
issue.

Thanks Again. I was not expecting this much of fast resolution. It's really
great.

Regards,
Jiten
